install/0000_90_machine-config-operator_01_prometheus-rules.yaml

```diff
@@ -93,13 +93,62 @@ spec:
             severity: warning
           annotations:
             message: "System memory usage of {{ $value | humanize }} on {{ $labels.node }} exceeds 95% of the reservation. Reserved memory ensures system processes can function even when the node is fully allocated and protects against workload out of memory events impacting the proper functioning of the node. The default reservation is expected to be sufficient for most configurations and should be increased (https://docs.openshift.com/container-platform/latest/nodes/nodes/nodes-nodes-managing.html) when running nodes with high numbers of pods (either due to rate of change or at steady state)."
-    - name: master-nodes-high-memory-usage
+    - name: high-overall-control-plane-memory
       rules:
-        - alert: MasterNodesHighMemoryUsage
+        - alert: HighOverallControlPlaneMemory
           expr: |
-            ((sum(node_memory_MemTotal_bytes AND on (instance) label_replace( kube_node_role{role="master"}, "instance", "$1", "node", "(.+)" )) - sum(node_memory_MemFree_bytes + node_memory_Buffers_bytes + node_memory_Cached_bytes AND on (instance) label_replace( kube_node_role{role="master"}, "instance", "$1", "node", "(.+)" ))) / sum(node_memory_MemTotal_bytes AND on (instance) label_replace( kube_node_role{role="master"}, "instance", "$1", "node", "(.+)" )) * 100) > 90
-          for: 15m
+            (
+              1
+              -
+              sum(
+                node_memory_MemFree_bytes
+                + node_memory_Buffers_bytes
+                + node_memory_Cached_bytes
+                AND on (instance)
+                label_replace( kube_node_role{role="master"}, "instance", "$1", "node", "(.+)" )
+              ) / sum(
+                node_memory_MemTotal_bytes
+                AND on (instance)
+                label_replace( kube_node_role{role="master"}, "instance", "$1", "node", "(.+)" )
+              )
+            ) * 100 > 60
```

Comment:
alerting if control plane MEM consumption is over 60%? that seems really low. I would push to 90.

Reply:
we (SRE) want to move away from alerts simply based on resource consumption. As mentioned https://issues.redhat.com/browse/OSD-9112 the use of resource consumption + increasing latency is the goal to indicate vertical scaling. These alerts/ideas are still immature - we could look to update docs once we have more data driven cases.

Reply:
I like alerting about user-visible impacts in most cases.  But in this case, the recommended response is "scale up the control plane", which today begins with "drain and shut down one of the instances, so we can make it bigger".  And we will also be draining and restarting the instance on many MachineConfig bumps.  Until we can surge new control-plane capacity in both of those cases, you want a lot of headroom.  If folks only look at these once the headroom is gone, things like control-plane updates and scale-ups will be more painful than they would have been had you come into them with something closer to our documented headroom suggestions.